x/incentive/simulation/operations/msg.go

```diff
@@ -0,0 +1,116 @@
+package operations
+
+import (
+	"fmt"
+	"math/rand"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	authexported "github.com/cosmos/cosmos-sdk/x/auth/exported"
+	"github.com/cosmos/cosmos-sdk/x/auth/vesting"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+	"github.com/cosmos/cosmos-sdk/x/supply"
+
+	"github.com/kava-labs/kava/x/incentive"
+	"github.com/kava-labs/kava/x/incentive/keeper"
+	"github.com/kava-labs/kava/x/incentive/types"
+	"github.com/kava-labs/kava/x/kavadist"
+)
+
+var (
+	noOpMsg = simulation.NoOpMsg(incentive.ModuleName)
+)
+
+// SimulateMsgClaimReward generates a MsgClaimReward
+func SimulateMsgClaimReward(ak auth.AccountKeeper, sk supply.Keeper, k keeper.Keeper) simulation.Operation {
+	handler := incentive.NewHandler(k)
+
+	return func(r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simulation.Account) (
+		simulation.OperationMsg, []simulation.FutureOperation, error) {
+
+		// Load only account types that can claim rewards
+		var accounts []authexported.Account
+		for _, acc := range accs {
+			account := ak.GetAccount(ctx, acc.Address)
+			switch account.(type) {
+			case *vesting.PeriodicVestingAccount, *auth.BaseAccount: // Valid: BaseAccount, PeriodicVestingAccount
+				accounts = append(accounts, account)
+				break
+			default: // Invalid: ValidatorVestingAccount, DelayedVestingAccount, ContinuousVestingAccount
+				break
+			}
+		}
+
+		// Load open claims and shuffle them to randomize
+		openClaims := types.Claims{}
+		k.IterateClaims(ctx, func(claim types.Claim) bool {
+			openClaims = append(openClaims, claim)
+			return false
+		})
+		r.Shuffle(len(openClaims), func(i, j int) {
+			openClaims[i], openClaims[j] = openClaims[j], openClaims[i]
+		})
+
+		// Randomly select a reward's collateral type from params
+		params := k.GetParams(ctx)
+		if len(params.Rewards) == 0 {
+			return noOpMsg, nil, fmt.Errorf("no rewards found in incentive module params")
+		}
+		reward := params.Rewards[r.Intn(len(params.Rewards))]
+
+		// Load kavadist module account's current balance
+		kavadistMacc := sk.GetModuleAccount(ctx, kavadist.KavaDistMacc)
+		kavadistBalance := kavadistMacc.SpendableCoins(ctx.BlockTime())
+
+		// Find address that has a claim of the same reward denom, then confirm it's distributable
+		claimer, claim, found := findValidAccountClaimPair(accounts, openClaims, func(acc authexported.Account, claim types.Claim) bool {
+			if claim.Owner.Equals(acc.GetAddress()) { // Account must be claim owner
+				if claim.Denom == reward.Denom { // Claim's denom must match our reward's denom
```

Comment:
see above, can skip this check

Reply:
https://github.com/Kava-Labs/kava/pull/439/commits/b47bb2f59a9bd19d17a0f6e6e0d4946dfb4b79aa